realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy

```diff
@@ -44,10 +42,42 @@ import static com.android.build.api.transform.QualifiedContent.*
 class RealmTransformer extends Transform {
 
     private Logger logger = LoggerFactory.getLogger('realm-logger')
-    private Project project
+    TransformerConfigProvider provider
 
-    public RealmTransformer(Project project) {
-        this.project = project
+    public RealmTransformer(TransformerConfigProvider provider) {
+        this.provider = provider
+    }
+
+    // constructor for OkBuck
+    public RealmTransformer(File configFile) {
+        if (!configFile.exists()) {
+            throw new FileNotFoundException(configFile.absolutePath)
+        }
+
+        Properties properties = new Properties()
+        configFile.withInputStream {
+            properties.load(it)
+        }
+        boolean syncEnabled = Boolean.valueOf(properties.getProperty("realm.syncEnabled", "false"))
+
+        def bootClassPath = properties.getProperty("realm.bootClassPath")
+        List<String> bootClassPathList = []
+        if (!(bootClassPath == null || bootClassPath.empty)) {
+            bootClassPath.split(',').each {
+                bootClassPathList.add(it.trim())
+            }
+        }
+        provider = new TransformerConfigProvider() {
+            @Override
+            boolean getSyncEnabled() {
+                return syncEnabled
+            }
+
+            @Override
+            List<String> getBootClassPathList() {
+                return bootClassPathList
```

Comment:
The boot classpath is already passed in the transform invocation as referenced input jar, so you just need to add that to your class loader.